boxsdk/object/base_api_json_object.py

```diff
@@ -0,0 +1,64 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+from abc import ABCMeta
+
+import six
+
+from boxsdk.util.translator import Translator
+
+
+class BaseAPIJSONObjectMeta(ABCMeta):
+    """
+    Metaclass for Box API objects. Registers classes so that API responses can be translated to the correct type.
+    Relies on the _item_type field defined on the classes to match the type property of the response json.
+    But the type-class mapping will only be registered if the module of the class is imported.
+    So it's also important to add the module name to __all__ in object/__init__.py.
+    """
+    def __init__(cls, name, bases, attrs):
+        super(BaseAPIJSONObjectMeta, cls).__init__(name, bases, attrs)
+        item_type = attrs.get('_item_type', None)
+        if item_type is not None:
+            Translator().register(item_type, cls)
+
+
+@six.add_metaclass(BaseAPIJSONObjectMeta)
+class BaseAPIJSONObject(object):
+    """Base class containing basic logic shared between true REST objects and other objects (such as an Event)"""
+
+    def __init__(self, object_id=None, response_object=None):
```

Comment:
Just like you got rid of `session` here, I think we can also get rid of `object_id`. Non-REST objects won't tend to have an `object_id`, so I think it'd be fine to omit it. That functionality can be directly on `BaseObject`, which is the only subclass that needs it.

Don't forget a docstring for this.

And since we can't guarantee the existence of an `object_id`, I think that we probably do want to require the existence of the `response_object` (I don't think we'll be creating `BaseAPIJSONObject` objects without them, unless they are `BaseObject`s). So I think we can make the signature be

```python
def __init__(self, response_object):
```